### PR TITLE
Add big uint type and use it for difficulty

### LINKED ISSUE #94: Large difficulty value breaks block serialization
it's 32 bits for root and 64 bit for minor block.
need to settle on the right value size for storing difficulty and make sure the difficulty will not overflow.

### PR DESCRIPTION
**Pull request by ninjaahhh**
fixes #94

### CODE REVIEW

File: quarkchain/core.py [hunk 1]
```diff
@@ -167,6 +167,25 @@ def deserialize(self, bb):
         return [self.ser.deserialize(bb) for i in range(size)]
 
 
+class BigUintSerializer:
+    """Not infinity, but pretty big: 2^2048-1"""
+
+    def __init__(self):
+        self.ser = PrependedSizeBytesSerializer(1)
```
Comment (qcgg) [inline quarkchain/core.py]: not used

File: quarkchain/tests/test_core.py [hunk 2]
```diff
@@ -108,3 +119,25 @@ def test_optional(self):
...
+
+
+class ABigUint(Serializable):
+    FIELDS = [("value", biguint)]
+
+    def __init__(self, value: int):
+        self.value = value
+
+
+class TestBigUint(unittest.TestCase):
```
Comment (qcgg) [inline quarkchain/tests/test_core.py]: test boundary values like 2 ** (255 * 8) - 1 and 2 ** (255 * 8) - 1  and 0?

File: quarkchain/core.py (hunk 1 above)
Comment (ninjaahhh) [reply quarkchain/core.py]: lol yeah I probbaly should simply use it

File: quarkchain/tests/test_core.py (hunk 2 above)
Comment (ninjaahhh) [reply quarkchain/tests/test_core.py]: done

File: quarkchain/core.py (hunk 1 above)
Comment (ninjaahhh) [reply quarkchain/core.py]: done
